dapps/marketplace/src/pages/nav/Profile.js

```diff
@@ -73,207 +91,332 @@ class ProfileNav extends Component {
-    {({ data, error }) => {
-      if (error) return null
-      const profile = get(data, 'web3.account.identity') || {}
-
-      return (
-        <div className="identity">
-          <h5>
-            <fbt desc="nav.profile.myIdentity">My Identity</fbt>
-          </h5>
-          <div className="info">
-            <Avatar profile={profile} size="3rem" />
-            <div>
-              <div className="name">
-                {profile.fullName ||
-                  fbt('Unnamed User', 'nav.profile.unnamedUser')}
-              </div>
-              <Attestations profile={profile} />
-            </div>
-          </div>
-          <div className="strength">
-            <div className="progress">
-              <div
-                className="progress-bar"
-                style={{ width: `${profile.strength || '0'}%` }}
+const ProfileDropdownRaw = ({
+  data,
+  identity,
+  identityLoading,
+  walletType,
+  // wallet,
+  // walletProxy,
+  // config,
+  onClose
+}) => {
+  const { checksumAddress, id } = data.web3.primaryAccount
+
+  return (
+    <div className="dropdown-menu dark dropdown-menu-right show profile">
+      <div className="active-wallet-info">
+        <Network />
+        <WalletAddress wallet={checksumAddress} walletType={walletType} />
+        {/* {!config.proxyAccountsEnabled ? null : (
+          <>
+            {walletProxy === wallet ? (
+              <DeployProxy
+                className="btn btn-sm btn-outline-primary px-3"
+                children="Deploy"
               />
-            </div>
-            {`${fbt(
-              'Profile Strength',
-              'nav.profile.ProfileStrength'
-            )} - ${profile.strength || '0'}%`}
-          </div>
-        </div>
-      )
-    }}
-  </Query>
-))
+            ) : (
+              <WalletAddress wallet={walletProxy} walletType={walletType}>
+                <fbt desc="nav.profile.proxyAccount">Proxy Account</fbt>
+              </WalletAddress>
+            )}
+          </>
+        )} */}
+      </div>
+      <div className="identity-info">
+        <Identity
+          id={id}
+          identity={identity}
+          identityLoading={identityLoading}
+          onClose={onClose}
+        />
+      </div>
+    </div>
+  )
+}
 
-export default ProfileNav
+const ProfileDropdown = withConfig(withWallet(ProfileDropdownRaw))
+
+function getWalletIconClass(walletType) {
+  switch (walletType) {
+    case 'Origin Wallet':
+      return 'origin'
+
+    case 'MetaMask':
+    case 'Meta Mask':
+      return 'metamask'
+
+    case 'Trust Wallet':
+      return 'trust'
+
+    case 'Coinbase Wallet':
+      return 'toshi'
+
+    case 'Cipher':
+      return 'cipher'
+
+    case 'Mist':
+      return 'mist'
+
+    case 'Parity':
+      return 'parity'
+  }
+
+  return 'metamask'
+}
+
+export default withWallet(withIdentity(ProfileNav))
 
 require('react-styl')(`
+  #app
+    overflow: hidden !important
```

Comment:
seems like a hack... is there no way to find out what's causing the overflow?

Reply:
Will look into it again

Reply:
The callout arrow was still being rendered despite having 0 height and width. This fixed it: https://github.com/OriginProtocol/origin/pull/2308/commits/456a1fe41ae4896f6fe6b82640a19a1c3c99ee11